crates/bevy_app/src/app.rs

```diff
@@ -129,6 +133,9 @@ impl App {
         let _bevy_app_run_guard = bevy_app_run_span.enter();
 
         let mut app = std::mem::replace(self, App::empty());
+        if app.is_from_plugin_build {
+            panic!("App::Run() is called from a Plugin::Build(), which might result in other initialization code not run.");
```

Comment:
```suggestion
            panic!("App::run() was called from within Plugin::build(), which might result in other initialization code not being run.");
```

Reply:
nit: Since this is a panic, I think this message will be slightly weird for users to encounter. 

More direct would be simply 
```
App::run() was called from within Plugin::build(), which is not allowed. 
```
This unambiguously tells the user the path forward, instead of implying there's some way trying to do this could be valid.

Reply:
I agree so I did that with https://github.com/bevyengine/bevy/commit/00ebeecb8b0ea83d49a5a370f318910d42ad7787

Reply:
Oh no it's not exactly the same sentence I was looking from mobile 😳 👍